rust/cubesql/cubesql/src/compile/engine/information_schema/postgres/pg_collation.rs

```diff
@@ -0,0 +1,259 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use datafusion::{
+    arrow::{
+        array::{Array, ArrayRef, BooleanBuilder, Int32Builder, StringBuilder, UInt32Builder},
+        datatypes::{DataType, Field, Schema},
+        record_batch::RecordBatch,
+    },
+    datasource::{datasource::TableProviderFilterPushDown, TableProvider},
+    error::Result,
+    logical_plan::Expr,
+    physical_plan::{memory::MemoryExec, ExecutionPlan},
+};
+
+use crate::compile::engine::information_schema::postgres::PG_NAMESPACE_CATALOG_OID;
+
+struct PgCollation {
+    oid: u32,
+    collname: &'static str,
+    collnamespace: u32,
+    collowner: u32,
+    collprovider: String,
+    collisdeterministic: bool,
+    collencoding: i32,
+    collcollate: Option<String>,
+    collctype: Option<String>,
+    // Column `colliculocale` renamed to `colllocale` since PostgreSQL 17.
+    // Support both columns for backward-compatibility.
+    // Reference: https://pgpedia.info/p/pg_collation.html
+    colliculocale: Option<String>,
+    colllocale: Option<String>,
+    collicurules: Option<String>,
+    collversion: Option<String>,
+}
+
+struct PgCatalogCollationBuilder {
+    oid: UInt32Builder,
+    collname: StringBuilder,
+    collnamespace: UInt32Builder,
+    collowner: UInt32Builder,
+    collprovider: StringBuilder,
+    collisdeterministic: BooleanBuilder,
+    collencoding: Int32Builder,
+    colliculocale: StringBuilder,
```

Comment:
With the suggestion above, it's a good idea to repeat the comment here.
```suggestion
    // Column `colliculocale` renamed to `colllocale` since PostgreSQL 17.
    // Support both columns for backward-compatibility.
    // Reference: https://pgpedia.info/p/pg_collation.html
    colliculocale: StringBuilder,
```